src/ast/nodes/ChainExpression.ts

```diff
@@ -1,9 +1,50 @@
+import type MagicString from 'magic-string';
+import type { RenderOptions } from '../../utils/renderHelpers';
+import type { DeoptimizableEntity } from '../DeoptimizableEntity';
+import type { HasEffectsContext } from '../ExecutionContext';
+import { SHARED_RECURSION_TRACKER } from '../utils/PathTracker';
+import { EMPTY_PATH } from '../utils/PathTracker';
 import type CallExpression from './CallExpression';
 import type MemberExpression from './MemberExpression';
 import type * as NodeType from './NodeType';
+import type { LiteralValueOrUnknown } from './shared/Expression';
+import { UnknownValue } from './shared/Expression';
 import { NodeBase } from './shared/Node';
 
-export default class ChainExpression extends NodeBase {
+const unset = Symbol('unset');
+
+export default class ChainExpression extends NodeBase implements DeoptimizableEntity {
 	declare expression: CallExpression | MemberExpression;
 	declare type: NodeType.tChainExpression;
+	private objectValue: LiteralValueOrUnknown | typeof unset = unset;
+
+	deoptimizeCache(): void {
+		this.objectValue = UnknownValue;
+	}
+
+	getLiteralValueAtPath(): LiteralValueOrUnknown {
+		if (this.getObjectValue() == null) return undefined;
+		return UnknownValue;
```

Comment:
This one should be easy to cover in a test, I wonder why you current test does not hit this line